pkgs/by-name/as/astro-language-server/package.nix

```diff
@@ -0,0 +1,102 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  pnpm_8,
+  nodejs_22,
+  jq,
+  moreutils,
+  cacert,
+  stdenvNoCC,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "astro-language-server";
+  version = "2.10.0";
+
+  src = fetchFromGitHub {
+    owner = "withastro";
+    repo = "language-tools";
+    rev = "@astrojs/language-server@${finalAttrs.version}";
+    hash = "sha256-WdeQQaC9AVHT+/pXLzaC6MZ6ddHsFSpxoDPHqWvqmiQ=";
+  };
+
+  pnpmDeps = stdenvNoCC.mkDerivation {
+    pname = "${finalAttrs.pname}-pnpm-deps";
+    inherit (finalAttrs) src version;
+
+    nativeBuildInputs = [
+      jq
+      moreutils
+      pnpm_8
+      cacert
+    ];
+
+    # https://github.com/NixOS/nixpkgs/blob/de80f1eeac3152c5bbfb1f8891b6414d526bfc54/pkgs/by-name/po/pot/package.nix#L54
+    installPhase = ''
+      export HOME=$(mktemp -d)
+      pnpm config set store-dir $out
+      pnpm config set side-effects-cache false
+      pnpm config set update-notifier false
+      pnpm config set dedupe-peer-dependents false
+      pnpm install --filter=@astrojs/language-server --frozen-lockfile --no-optional --ignore-script --force
```

Comment:
The filter thing is important, but also setting dedupe-peer-dependants above, which is required for this package to build properly.

Reply:
> The filter thing is important, but also setting dedupe-peer-dependants above, which is required for this package to build properly.

Hmm it'd be nice to make `pnpm.fetchDeps` accept arguments that would allow to add arbitrarily such config options.. cc @Scrumplex